calculator/src/Calculator.js

```diff
@@ -0,0 +1,171 @@
+import { Component } from 'react';
+import './css/index.css';
+import {
+  FONT_SIZE_STANDARD,
+  INFINITY_MESSAGE,
+  MAX_CURRENT_LENGTH,
+} from './constants';
+
+class Calculator extends Component {
+  constructor() {
+    super();
+
+    this.state = JSON.parse(localStorage.getItem('state')) ?? {
+      total: 0,
+      current: 0,
+      operator: '',
+      isLastClickOperator: false,
+    };
+  }
+
+  componentDidUpdate() {
+    localStorage.setItem('state', JSON.stringify(this.state));
+  }
+
+  handleDigitClick(digitValue) {
+    const { current, isLastClickOperator } = this.state;
+
+    if (current.toString().length > MAX_CURRENT_LENGTH) {
+      this.setState({ current: Infinity });
+      return;
+    }
+
+    if (isLastClickOperator) {
+      this.enterNewOperand(digitValue);
+    }
+
+    if (!isLastClickOperator) {
+      this.concatDigitOperand(digitValue);
+    }
+  }
+
+  enterNewOperand(digitValue) {
+    this.setState({
+      current: digitValue,
+      isLastClickOperator: false,
+    });
+  }
+
+  concatDigitOperand(digitValue) {
+    const { current } = this.state;
+
+    this.setState({
+      current: current * 10 + digitValue,
+    });
+  }
+
+  handleOperatorClick(operatorValue) {
+    const { isLastClickOperator, operator, total, current } = this.state;
+
+    this.updateOperator(operatorValue);
+
+    if (isLastClickOperator) return;
+
+    if (!this.isOperatorExist()) {
+      this.calculate(current);
+    }
+
+    if (this.isOperatorExist()) {
+      const result = this.operate(total, current, operator);
+
+      this.calculate(result);
+    }
```

Comment:
```suggestion
    if (!this.isOperatorExist()) {
      this.calculate(current);
      return;
    }

    const result = this.operate(total, current, operator);

    this.calculate(result);
```
return 하면서 아래 조건문을 두지 않을 수 있겠네요 🙂

Reply:
early return 또는 삼항 연산자를 활용해서 리팩토링할까 고민했었는데요. 코드 이해가 쉽지 않을 것 같아서 고치지 않았습니다 ㅠ 

```js
 this.calculate(
      this.isOperatorExist() ? this.operate(total, current, operator) : current
 );
```

사실 위와 같이 고치는 것이 가장 간단해보이기는 하는데 코드를 처음 접하는 사람이 이해하기 쉬울지 의문입니다 ㅠ 
율리님의 생각은 어떠신가요?

Reply:
음- 해당 조건이 아닌 경우 다음 코드 흐름으로 넘어가는거니까 그렇게 코드 이해가 어렵다고 느끼진 않을것 같네요. 🙂 그리고 보통은 삼항연산자보다 if문을 이용하는것이 더 가독성이 좋은 편이라고 느껴져요. 

 조건문 외에도 이 내부 코드를 보는 입장에서 잘 이해할수 있도록 좀 더 고민해 보셨으면 좋겠다고 생각한 부분에 대해 말씀드려보자면 operate 와 calculate 메소드명이 유사하게 느껴져서 역할을 바로 알기 힘들었어요. operate가 calculate라는 이름이 더 어울리고 calculate 는 updateCurrent(또는 current, total 상태를 업데이트 한다는 네이밍)이 더 어울리지 않을까하는데 무비는 어떻게 생각할지 궁금하네요~ 🤔

Reply:
헉 맞아요. 사실 `operate`와 `calculate` 둘의 이름이 너무 혼동돼서 이해하기가 어려웠던 것 같아요! 
조건문만의 문제가 아니였군요.! 

율리님의 말씀이 맞는 것 같아요. 
계산하는 메소드 이름은 `calculate`,
상태를 변경해주는 메소드 이름은 `updateResult`로 변경해주었습니다.

이름을 바꿔주니 코드이해가 더 쉬워진 것 같아요!

[refactor: change method name](https://github.com/byhhh2/react-calculator/commit/457194e47c90c97c96fe06211bae26ca96f9933e)
[refactor: change logic](https://github.com/byhhh2/react-calculator/commit/dbe6290cf558feb31833798bd7c4f43fea9cde91)